R/dataprep_table_formatting.R

```diff
@@ -0,0 +1,80 @@
+#' as_update_existing
+#' 
+#' Converts a sidora table to a format that can be use with pandora table upload
+#' interface
+#' 
+#' @examples
+#' \dontrun{
+#' sidora_table <- get_df("TAB_Site", con)
+#' format_as_update_existing(sidora_table,)
+#' }
+#' 
+#' @param sidora_table a sidora table
+#' 
+#' @rdname format_as_update_existing
+#' @export
+format_as_update_existing <- function(sidora_table) {
+  
+  ## Find valid columns for uploading existing tables
+  valid_cols_table_clean <- valid_cols_table <- hash::values(hash_pandora_col_name_update_type, colnames(sidora_table))
+  names(valid_cols_table_clean) <- sidora_col_name_to_update_col_name(names(valid_cols_table_clean))
+  valid_cols_table_clean_mandatory <- names(valid_cols_table_clean[valid_cols_table_clean %in% "mandatory"])
+  
+  ## For upload 'Full_*_Id' column isn't used, but just '_Id', which here is a 
+  ## numeric value . We will replace the latter with the former here for export
+  ## QUESTION maybe put as data object? But will never be used other than here
```

Comment:
Couldn't you just gsub the "Full_"? away?

Reply:
I could but it would be basically the same number of lines, and less robust

What you propose: 

```
name_map <- c("individual.Individual_Id", 
                "sample.Sample_Id", 
                "extract.Extract_Id", 
                "library.Library_Id", 
                "capture.Capture_Id", 
                "sequencing.Sequencing_Id", 
                "raw_data.Raw_Data_Id"
                )
  
  sidora_table <- sidora_table[!colnames(sidora_table) %in% name_map]
  colnames(sidora_table) <- gsub(".Full", "", colnames(sidora_table))
```
But the challenge is that it creates duplicate column name (`indiviudal.Individual_Id`, `Individual.Full_Individual_Id` -> `indiviudal.Individual_Id`, `Individual.Individual_Id`). Therefore I have to remove the original numeric Id column. 

But then to use regex to remove this original column (e.g. `individual.Individual_Id`), without removing other columns with the `_Id` suffix (e.g. `sample.C14_Id`) becomes very complicated.

It is also less robust if in other columns with `.Full_` are added in the future.

Therefore I prefer to duplicate the data using an explicit 'map' of columns to modify rather than removing stuff in this case.

Reply:
Alright - good call!